src/Nethermind/Nethermind.Serialization.FluentRlp/Int32Primitive.cs

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers.Binary;
+
+namespace Nethermind.Serialization.FluentRlp;
+
+internal static class Int32Primitive
+{
+    /// <summary>
+    /// Reads a <see cref="int" /> from the beginning of a read-only span of bytes, as big endian.
+    /// </summary>
+    /// <param name="source">The read-only span to read.</param>
+    /// <returns>The big endian value.</returns>
+    /// <remarks>The span is padded with leading `0`s as needed.</remarks>
+    public static int Read(ReadOnlySpan<byte> source)
+    {
+        Span<byte> buffer = stackalloc byte[sizeof(Int32)];
+        source.CopyTo(buffer[^source.Length..]);
```

Comment:
`Unsafe.ReadUnalaligned`? Why copy?

Reply:
This method was added due to `BinaryPrimitives.ReadInt32BigEndian` requiring exactly 4 byte to read a `Int32`, so we pad `source` with enough `0` so it can properly decode.

I've never used `Unsafe.ReadUnaligned`, not even `usafe` code. How would that look?

Reply:
Why not use `Bytes` class we already have?

Reply:
Didn't think about it, but seems like implementations are the same:

https://github.com/NethermindEth/nethermind/blob/c8748770fe88fecf3b0b36226296e5c0858da051/src/Nethermind/Nethermind.Core/Extensions/Bytes.cs#L439-L441

Seems a bit overkill to add a project reference for 3 LOCs.